summertime/model/single_doc/multilingual/translation_pipeline_model.py

```diff
@@ -0,0 +1,148 @@
+from .base_multilingual_model import MultilingualSummModel, fasttext_predict
+from summertime.model.base_model import SummModel
+from summertime.model.single_doc import BartModel
+
+from easynmt import EasyNMT
+
+
+class TranslationPipelineModel(MultilingualSummModel):
+    """
+    A class for multilingual summarization performed by first
+    translating into English then performing summarization in English.
+    """
+
+    model_name = "Translation Pipeline"
+    is_multilingual = True
+    # TODO: change to Pegasus as default?
+    # language codes from https://github.com/UKPLab/EasyNMT#Opus-MT documentation
+    # language codes not supported by https://fasttext.cc/docs/en/language-identification.html
+    # are removed
+    supported_langs = [
+        "aed",
+        "af",
+        "am",
+        "ar",
+        "az",
+        "bat",
+        "bcl",
+        "be",
+        "bg",
+        "bn",
+        "ca",
+        "ceb",
+        "cs",
+        "cy",
+        "da",
+        "de",
+        "el",
+        "en",
+        "eo",
+        "es",
+        "et",
+        "eu",
+        "fi",
+        "fr",
+        "ga",
+        "gl",
+        "gv",
+        "he",
+        "hi",
+        "hr",
+        "ht",
+        "hu",
+        "hy",
+        "id",
+        "ilo",
+        "is",
+        "it",
+        "ja",
+        "ka",
+        "ko",
+        "lt",
+        "lv",
+        "mg",
+        "mk",
+        "ml",
+        "mr",
+        "ms",
+        "mt",
+        "nl",
+        "no",
+        "pa",
+        "pl",
+        "pt",
+        "ro",
+        "ru",
+        "run",
+        "sh",
+        "sk",
+        "sl",
+        "sq",
+        "sv",
+        "sw",
+        "th",
+        "tl",
+        "tr",
+        "uk",
+        "ur",
+        "vi",
+        "wa",
+        "war",
+        "yo",
+        "zh",
+    ]
+
+    lang_tag_dict = {lang: lang for lang in supported_langs}
+
+    def __init__(self, model_backend: SummModel = BartModel, **kwargs):
+
+        model: SummModel = model_backend(**kwargs)
+        self.model = model
+
+        super(TranslationPipelineModel, self).__init__(
+            trained_domain=self.model.trained_domain,
+            max_input_length=self.model.max_input_length,
+            max_output_length=self.model.max_output_length,
+        )
+
+        # translation module
+        self.translator = EasyNMT("opus-mt")
+
+    def summarize(self, corpus, queries=None):
+        self.assert_summ_input_type(corpus, queries)
+
+        src_lang = fasttext_predict(corpus)
+        # translate to English
+        corpus = self.translator.translate(
+            corpus, source_lang=src_lang, target_lang="en", beam_size=4
+        )
+        # TODO: translate each doc separately if provided multiple docs in corpus?
+        if queries:
+            queries = self.translator.translate(queries, target_lang="en", beam_size=4)
```

Comment:
when you specify the beam_size as 4, is the output one sentence or 4?

Reply:
Still just one sentence -- this is just the beam size for the translation model's generation.